pyriemann/datasets/sampling.py

```diff
@@ -0,0 +1,338 @@
+        Gaussian distributions on the space of symmetric positive definite
+        matrices”, IEEE Trans Inf Theory, vol. 63, pp. 2153–2170, 2017.
+        https://arxiv.org/pdf/1507.01760.pdf
+    """
+
+    samples_r = _sample_parameter_r(n_samples=n_samples,
+                                    n_dim=n_dim,
+                                    sigma=sigma,
+                                    random_state=random_state)
+    samples_U = _sample_parameter_U(n_samples=n_samples,
+                                    n_dim=n_dim,
+                                    random_state=random_state)
+
+    samples = np.zeros((n_samples, n_dim, n_dim))
+    for i in range(n_samples):
+        Ui = samples_U[i]
+        ri = samples_r[i]
+        samples[i] = Ui.T @ np.diag(np.exp(ri)) @ Ui
+        samples[i] = (samples[i] + samples[i].T) / 2.0  # ensure symmetry
+
+    return samples
+
+
+def sample_gaussian_spd(n_samples, mean, sigma, random_state=None):
+    """Sample a Riemannian Gaussian distribution
+
+    Sample SPD matrices from a Riemannian Gaussian distribution centered at
+    mean and with dispersion parametrized by sigma. This distribution has been
+    defined in [1]_ and generalizes the notion of a Gaussian distribution to
+    the space of SPD matrices. The sampling is based on a spectral
+    factorization of SPD matrices in terms of their eigenvectors (U-parameters)
+    and the log of the eigenvalues (r-parameters).
+
+    Parameters
+    ----------
+    n_samples : int
+        How many samples to generate.
+    mean : ndarray, shape (n_dim, n_dim)
+        Center of the Riemannian Gaussian distribution.
+    sigma : float
+        Dispersion of the Riemannian Gaussian distribution.
+    random_state : int, RandomState instance or None (default: None)
+        Pass an int for reproducible output across multiple function calls.
+
+    Returns
+    -------
+    samples : ndarray, shape (n_samples, n_dim, n_dim)
+        Samples of the Riemannian Gaussian distribution
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+
+    References
+    ----------
+    .. [1] S. Said, L. Bombrun, Y. Berthoumieu, and J. Manton, “Riemannian
+        Gaussian distributions on the space of symmetric positive definite
+        matrices”, IEEE Trans Inf Theory, vol. 63, pp. 2153–2170, 2017.
+        https://arxiv.org/pdf/1507.01760.pdf
+    """
+
+    n_dim = mean.shape[0]
+    samples_centered = sample_gaussian_spd_centered(n_samples,
+                                                    n_dim=n_dim,
+                                                    sigma=sigma,
+                                                    random_state=random_state)
+
+    # apply the parallel transport to mean on each of the samples
+    samples = np.zeros((n_samples, n_dim, n_dim))
+    mean_sqrt = sqrtm(mean)
+    for i in range(n_samples):
+        samples[i] = mean_sqrt @ samples_centered[i] @ mean_sqrt
```

Comment:
I think that `samples = mean_sqrt @ samples_centered @ mean_sqrt` could speed up the code.